include/spm.h

```diff
@@ -0,0 +1,52 @@
+/*
```

Comment:
We need conceptual documentation for this as well - should this be part of this PR or does it come later?

Reply:
My thought was that we only have to move/rename the documentation, since this functionality already was in place, and documented. 

If new conceptual documentation is needed, it should be part of this PR IMO.

Reply:
spm.h is a new library - before we had only the sample. I assume people could use this library in their own application, so we'll need some documentation for it. Explaining what the library is for, how it can be used, pointing to the example, and including the API documentation.

Reply:
Got it. Should I just look at some other library documentation and try to write something similar for `spm.h` ?

Reply:
yes, please :)